BlueprintUI/Sources/ViewDescription/LifecycleCallback.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+/// This is the type used by all lifecycle callback hooks.
+///
+/// - Tag: ElementLifecycle
+///
+/// # The Element Lifecycle
+///
+/// View-backed elements have a lifecycle with 3 states:
+///
+/// 1. **Unmounted**
+///
+///    This is the initial state of an element, before a render pass.
+///
+/// 2. **Mounted**
+///
+///    When an element is present during a render pass, it is "mounted". This means the element's
+///    backing view has been created and added to the hosting `BlueprintView`. However, the
+///    element's backing view may not yet be visible, if the hosting view is not part of the view
+///    hierarchy yet.
+///
+/// 3. **Visible**
+///
+///    Finally, when the hosting view is added to the view hierarchy, the element becomes visible.
+///    If the hosting view is already in a view hierarchy when the element is mounted, it will
+///    immediately transition to this state.
+///
+/// There are 4 lifecycle events that occur when an element transitions between these states. An
+/// element will always be mounted before it is visible, and will disappear before it is unmounted.
+///
+/// The following diagram shows how an element may transition between states:
+///
+/// ```
+///           ┌───onMount───┐   ┌──onAppear───┐
+///           │             │   │             │
+///   ┌───────┴───┐     ┌───▼───┴───┐     ┌───▼───────┐
+///   │ Unmounted │     │  Mounted  │     │  Visible  │
+///   └───────▲───┘     └───┬───▲───┘     └───┬───────┘
+///           │             │   │             │
+///           └──onUnmount──┘   └─onDisappear─┘
+/// ```
+///
+/// - Note: If a hosting view is deallocated while it contains mounted elements, those elements will
```

Comment:
Could/should we call `onUnmount` in the `BlueprintView`'s `deinit` (if it isn't already in the unmounted state)?

Reply:
> Could/should we call `onUnmount` in the `BlueprintView`'s `deinit` (if it isn't already in the unmounted state)?

Hmm, yeah, I think that's doable, if we want to guarantee  that final unmount.

Reply:
I agree, it seems nice to guarantee it

Reply:
OK, done.

Reply:
![image](https://user-images.githubusercontent.com/327847/130145554-510c838e-1eef-459a-a126-8c15d6bb73b5.png)